MigrationV3V4/Node/core-MultiDialogs-v4/dialogs/hotels.js

```diff
@@ -1,60 +1,85 @@
+const { ComponentDialog, WaterfallDialog } = require('botbuilder-dialogs');
 const { AttachmentLayoutTypes, CardFactory } = require('botbuilder');
 const store = require('../store');
 const { 
     INITIAL_HOTEL_PROMPT, 
     CHECKIN_DATETIME_PROMPT, 
-    HOW_MANY_NIGHTS_PROMPT 
+    HOW_MANY_NIGHTS_PROMPT,
+    HOTELS_WATERFALL_DIALOG
 } = require('../const');
 
-const hotelsDialog = [
-    async (stepContext) => {
+class HotelsDialog extends ComponentDialog {
+    constructor(id) {
+        super(id);
+
+        this.addDialog(new WaterfallDialog(HOTELS_WATERFALL_DIALOG, [
+            this.destinationPromptStep,
+            this.destinationSearchStep.bind(this),
+            this.checkinPromptStep.bind(this),
+            this.checkinTimeSetStep.bind(this),
+            this.stayDurationPromptStep.bind(this),
+            this.stayDurationSetStep.bind(this),
+            this.hotelSearchStep.bind(this)
+        ]));
+
+        this.initialDialogId = HOTELS_WATERFALL_DIALOG;
+
+    }
+
+    async destinationPromptStep (stepContext) {
         await stepContext.context.sendActivity('Welcome to the Hotels finder!');
         return await stepContext.prompt(
             INITIAL_HOTEL_PROMPT, {
                 prompt: 'Please enter your destination'
             }
         );
-    },
-    async (stepContext) => {
+    } 
+
+    async destinationSearchStep (stepContext) {
         const destination = stepContext.result;
         stepContext.values.destination = destination;
         await stepContext.context.sendActivity(`Looking for hotels in ${destination}`);
         return stepContext.next();
-    },
-    async (stepContext) => {
+    } 
+
+    async checkinPromptStep (stepContext) {
         return await stepContext.prompt(
             CHECKIN_DATETIME_PROMPT, {
                 prompt: 'When do you want to check in?'
             }
         );
-    },
-    async (stepContext) => {
+    }
+
+    async checkinTimeSetStep (stepContext) {
         const checkinTime = stepContext.result[1].value;
```

Comment:
This should be stepContext.result[0].value;    (stepContext.result[1] is undefined)

Reply:
strange - i retested this and both index 0 and 1 contain the correct value. i must have been seeing something differently before to have used '1'. i'll change it to '0' for now.